Updated artillery behaviour to get rid of destroyer-capsule

Jayefuu opened the pull request:
Use of the destroyer-capsule projectile instead of destroyer capsule meant that the destroyers and defenders didn't attack player entities until they reached the destination. They also couldn't be shot down by a well defended wall, they'd fly over.

I changed to destroyer-capsule so that they couldn't pick the artillery up after the bots spawned as this would remove their target and cause them to float around.

By request I have reveret to the destroyers that can be shot down as they're more fun and at current balance the destroyers kill the artillery every time, which is not much fun when they cannot be defended against.

To mitigate against players picking up the artillery after the destroyer/defender spawns, artillery targets will become unminable for 5 seconds after the destroyer/defender deploys.

This leaves the bug where they can pick up the artillery before the last shell destroys the turret. I considered adding artillery to a table when placed and checking their ammo inventory once every few seconds. I was going to set all artillery turrets to unminable unless they have no ammo for 3 seconds, but this seems like a lot of work for a small bug.

Jayefuu committed:
Updated artillery behaviour to get rid of destroyer-capsule while avoiding part of pickup bug

grilledham commented on map_gen/maps/crash_site/events.lua:
```diff
@@ -318,6 +318,15 @@ destroyer_callback =
+local artillery_cooldown_callback
+artillery_cooldown_callback =
+    Token.register(
+    function(data)
```
`data` could just be the entity itself, don't have to put it in a table.

Jayefuu replied:
I've now updated this.

Jayefuu committed:
Updated as per suggestions

Jayefuu committed:
Updated pollution multiplier commands

- Changed so that both can be run from server
- Gave regulars the ability to query the current pollution settings.

Jayefuu commented:
Woops. I committed some changes to this branch I didn't mean to. Those need checking too @grilledham as I don't know how to uncommit them.

grilledham commented on map_gen/maps/crash_site/outpost_builder.lua:
```diff
@@ -1219,13 +1219,16 @@ local function do_artillery_turrets_targets()
         return
     end
 
+    player = player or server_player
```
This is too low, it would need to happen before `player.print` 4 lines up.

grilledham commented on map_gen/maps/crash_site/outpost_builder.lua:
```diff
@@ -1953,7 +1955,7 @@ Command.add(
     {
         description = {'command_description.get_pollution_multiplier'},
-        required_rank = Ranks.admin,
+        required_rank = Ranks.regular,
```
What is the value of restricting this to regulars? If we let regulars do this we might as well let everyone.

Jayefuu replied:
None. I've switched it to guest.

grilledham commented on map_gen/maps/crash_site/events.lua:
```diff
@@ -318,6 +318,14 @@ destroyer_callback =
+artillery_cooldown_callback =
+    Token.register(
+    function(entity)
+        entity.minable = true
```
Now that I think about this again we need to check the entity is still valid, it might have killed.

Jayefuu committed:
Updated as per review

Jayefuu committed:
LUA checks
